### PR TITLE
feat: Add lazy support to available items grid

### LINKED ISSUE #130: Add support for Lazy Data Provider
It'd be great if we could use lazy data providers, just like in regular grids.

### PR DESCRIPTION
**Pull request by flang**
Close #130

### CODE REVIEW

**Comment by javier-godoy** [comment]
Cherrypick https://github.com/FlowingCode/TwinColGridAddon/commit/95499801bf83392e3a5abc99cff0fc19383babb2 for fixing compilation with Vaadin 24 (#133)

File: src/test/java/com/flowingcode/vaadin/addons/twincolgrid/FilterableDemo.java [hunk 1]
```diff
@@ -20,18 +20,17 @@
...
 import com.vaadin.flow.router.Route;
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import org.apache.commons.lang3.StringUtils;
 
 @SuppressWarnings("serial")
 @PageTitle("Filterable")
-@DemoSource
```
Comment (paodb) [inline src/test/java/com/flowingcode/vaadin/addons/twincolgrid/FilterableDemo.java:34]: Is there a reason why @DemoSource is being removed here?

File: src/test/java/com/flowingcode/vaadin/addons/twincolgrid/LazyFilterableDemo.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import com.vaadin.flow.router.Route;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+
+@SuppressWarnings("serial")
+@PageTitle("Lazy Filterable")
+@Route(value = "twincolgrid/lazyfilterable", layout = TwincolDemoView.class)
```
Comment (paodb) [inline src/test/java/com/flowingcode/vaadin/addons/twincolgrid/LazyFilterableDemo.java:41]: @DemoSource is missing here

**Review by paodb — CHANGES_REQUESTED** [review]
See comments

**Pull request closed without merging**